numba/np/linalg.py

```diff
@@ -903,6 +904,142 @@ def impl(a):
     return impl
 
 
+@overload(np.vecdot)
+def np_vecdot(x1, x2):
+    return np_vecdot_impl(x1, x2)
+
+@overload(np.vecdot)
+def np_vecdot_axis(x1, x2, axis):
+    # Handle axis parameter - it might be a literal type
+    if not isinstance(axis, types.Literal):
+        def impl(x1, x2, axis):
+            return np_vecdot_impl(x1, x2, axis=literally(axis))
+        return impl
+    else:
+        print(x1, x2, axis)
+        return np_vecdot_impl(x1, x2, axis=axis.literal_value)
```

Comment:
This should be in a `def impl(x1, x2, axis): ...`

According to your traceback, it's calling into actual `numpy` instead of numba compiled code.

Note: the most confusing concept with `@overload` is the distinction between typing and lowering. `@overload` should return a function object that will be compiled. Otherwise (like this call), is triggering with type object instead of actual values.